charmcraft/utils.py

```diff
@@ -185,3 +186,26 @@ def get_host_architecture():
     """Get host architecture in deb format suitable for base definition."""
     os_platform = get_os_platform()
     return ARCH_TRANSLATIONS.get(os_platform.machine, os_platform.machine)
+
+
+def confirm_with_user(prompt, default=False) -> bool:
+    """Query user for yes/no answer.
+
+    If TTY is not interaction, returns default value.
+    If answer is empty (""), returns default value.
+
+    :returns: True if yes (Y,y,YES,yes), otherwise false.
+    """
+    if not sys.stdin.isatty():
```

Comment:
I would also raise RuntimeError if used in managed-mode for now, maybe worth considering or are most runs except for the shell related ones running with stdin closed? In that case, it could be the conditional following this one :-)

Reply:
Fair point, though this shouldn't be called in managed-mode.  But a guard would be nice.

Reply:
Done!